x-pack/plugins/security_solution/server/endpoint/routes/metadata/support/agent_status.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClientContract } from 'kibana/server';
+import { AgentService } from '../../../../../../ingest_manager/server';
+import { AgentStatusKueryHelper } from '../../../../../../ingest_manager/common/services';
+import { Agent } from '../../../../../../ingest_manager/common/types/models';
+
+const STATUS_QUERY_MAP = new Map([
+  ['online', AgentStatusKueryHelper.buildKueryForOnlineAgents()],
```

Comment:
You need the enum here too

Reply:
well now this is confusing.

The schema is arrayOf oneOf.. some string literals. Since i had to call `toString()` on them. So the request that comes in, the type is definitely `string[]`.  So I would have to translate from the `string` status, to the enum (using `HOST_STATUS_MAPPING[x]`) and then from the enum to the kuery here..

So it would be `STATUS_QUERY_MAP.get(HOST_STATUS_MAPPING[x])`. I don't know that that's any cleaner

Reply:
The map is a string to function. Am not sure I understand, perhaps I am not seeing something.

Reply:
map is string to string, the functions are evaluated immediately, once. They're really just string builders so it's not costly to do it at call time or up front.

but `status` as it comes into the function is `string[]`, so the key for this map can stay `string`.

Reply:
Since the key is string we want to use the Enum.toString so we have strong to the centrally placed string. Such that when something changes about the list of status we support we can do it in one place. Also the code breaks too if one of the status is dropped from our support.